location.go

```diff
@@ -35,3 +40,23 @@ func DefaultWindowLocation() Location {
 		Longitude: 172.6362,
 	}
 }
+
+func locationToMap(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
+	if t != reflect.TypeOf(map[string]interface{}{}) {
+		return data, nil
+	}
+	switch f {
+	case reflect.TypeOf(&Location{}):
+		var m map[string]interface{}
+		err := mapstructure.Decode(data, &m)
+		m["Timestamp"] = data.(*Location).Timestamp.Truncate(time.Second)
+		return m, err
+	case reflect.TypeOf(Location{}):
+		var m map[string]interface{}
+		err := mapstructure.Decode(data, &m)
+		m["Timestamp"] = data.(Location).Timestamp.Truncate(time.Second)
+		return m, err
+	default:
```

Comment:
You could avoid some duplication by normalising pointer values like this:

```go
case reflect.TypeOf(&Location{}), 
    data = *(data.(*Location))  // follow the pointer
    fallthrough
case reflect.TypeOf(Location{}):
    var m map[string]interface{}
    err := mapstructure.Decode(data, &m)
    m["Timestamp"] = data.(Location).Timestamp.Truncate(time.Second)
    return m, err
default:
    ...
```

I'm not sure if you have a similar pattern elsewhere in go-config. Could be worth updating them all.

Reply:
Just location for now. But I expect this will grow over time.